api/proposals.py

```diff
@@ -198,3 +201,31 @@ def delete_proposal_policies(node_key):
 
 def generate_etag(obj):
     return hashlib.md5(json.dumps(obj).encode("utf-8")).hexdigest()
+
+
+# process_node_availabilities processes node availability data
+# in batches and inserts to db. This works happens in separate thread.
+def process_node_availabilities(app):
+    def create_session():
+        session_factory = sessionmaker(bind=db.get_engine(app))
+        return scoped_session(session_factory)()
+
+    batch = []
+    max_size = 20
+    while True:
+        try:
+            item = node_availability_queue.get()
+            batch.append(item)
+            if len(batch) >= max_size:
+                db_session = create_session()
+                db_session.bulk_save_objects(batch)
+                db_session.commit()
+                db_session.close()
+                batch = []
```

Comment:
This is wrong. Queue should also be cleared but it cannot be cleared here as other requests may be writing to it. Check if there is a way to set max queue items.